content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/check_service_retired.rb

```diff
@@ -1,37 +1,71 @@
-#
-# Description: This method checks to see that all of the service resources are retired before retiring the service.
-#
-
-service = $evm.root['service']
-if service.nil?
-  $evm.log('error', "Service Object not found")
-  exit MIQ_ABORT
-end
+module ManageIQ
+  module Automate
+    module Service
+      module Retirement
+        module StateMachines
+          module Methods
+            class CheckServiceRetired
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                service = service_obj
+
+                @handle.log('info', "Checking if all service resources have been retired.")
+                result = 'ok'
+                service.service_resources.each do |sr|
+                  next if sr.resource.nil?
+                  next if sr.resource_type == "Service" && @handle.vmdb(:service).find(sr[:resource_id]).has_parent
+                  next unless sr.resource.respond_to?(:retired?)
+                  result = check_service_resource_retirement(sr, service)
```

Comment:
@d-m-u 
Since this is in a loop is the result changing for each iteration, what happens if the first one has an error and the second one has a success, we will report the success but not the first failure